nss/integration-tests/helper_test.go

```diff
@@ -44,44 +48,45 @@ func outNSSCommandForLib(t *testing.T, rootUID, rootGID, shadowMode int, cacheDi
 	return got, err
 }
 
-// createTempDir creates a temporary directory with a cleanup teardown not having a testing.T.
-func createTempDir() (tmp string, cleanup func(), err error) {
-	if tmp, err = os.MkdirTemp("", "aad-auth-integration-tests-nss"); err != nil {
-		fmt.Fprintf(os.Stderr, "Can not create temporary directory %q", tmp)
-		return "", nil, err
-	}
-	return tmp, func() {
-		if err := os.RemoveAll(tmp); err != nil {
-			fmt.Fprintf(os.Stderr, "Can not clean up temporary directory %q", tmp)
-		}
-	}, nil
-}
-
 // buildRustNSSLib builds the NSS library with the feature integration-tests enabled and copies the
 // compiled file to libPath.
-func buildRustNSSLib() error {
-	aadPath, err := filepath.Abs("../..")
-	if err != nil {
-		return err
-	}
-	// Builds the nss library.
-	args := []string{"build", "--verbose", "--features", "integration-tests", "--target-dir", targetDir}
+func buildRustNSSLib(t *testing.T) {
+	t.Helper()
+
+	// Gets the path to the integration-tests.
+	_, p, _, _ := runtime.Caller(0)
+	l := strings.Split(filepath.Dir(p), "/")
+	// Walk up the tree to get the path of the project root
+	aadPath := "/" + filepath.Join(l[:len(l)-2]...)
+
+	rustDir := filepath.Join(aadPath, "nss")
+	testutils.MarkRustFilesForTestCache(t, rustDir)
+	env, target := testutils.TrackRustCoverage(t, rustDir)
 
 	cargo := os.Getenv("CARGO_PATH")
 	if cargo == "" {
 		cargo = "cargo"
 	}
+
+	// Builds the nss library.
+	args := []string{"build", "--verbose", "--all-features", "--target-dir", target}
 	// #nosec:G204 - we control the command arguments in tests
 	cmd := exec.Command(cargo, args...)
+	cmd.Env = append(os.Environ(), env...)
 	cmd.Dir = aadPath
-	if out, err := cmd.CombinedOutput(); err != nil {
-		return fmt.Errorf("could not build rust nss library (%s): %w", out, err)
-	}
-	targetDir = filepath.Join(targetDir, os.Getenv("DEB_HOST_RUST_TYPE"))
 
-	// Renames the compiled library to have the expected versioned name.
-	if err = os.Rename(filepath.Join(targetDir, "debug", "libnss_aad.so"), libPath); err != nil {
-		return fmt.Errorf("Setup: could not rename the Rust NSS library: %w", err)
+	out, err := cmd.CombinedOutput()
+	require.NoError(t, err, "Setup: could not build Rust NSS library: %s", out)
+
+	if cargo != "cargo" {
```

Comment:
hum, maybe tell this is to differentiate the real binary from the debian wrapper  (and so, move your comment above). How does it relate to the current PR though?

Reply:
This was already there. I think it's just a diff weirdness, since I shifted a couple of things around to make it more clear

Reply:
I think it worthes a comment as least (because if you don’t know `dh_cargo` is using a wrapper script with a different path, you will not know why this check…)

Reply:
I agree, but the more I think about it, maybe it's better to remove this if and just leave the filepath.Join line (with some improvements on the rationale comment), since the CARGO_PATH env could still be used to set another cargo path than the one present in PATH.